placeholder-service/main/java/com/f4k0/placeholder/maps/MapsClient.java

```diff
@@ -0,0 +1,89 @@
+package com.f4k0.placeholder.maps;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.reactive.function.client.WebClient;
+import place.Place;
+import place.PlaceList;
+
+import java.util.Optional;
+
+public class MapsClient {
+
+    private static final int RADIUS = 500;
+    private static final Logger LOGGER = LoggerFactory.getLogger(MapsClient.class);
+    private static final String GMAPS_BASE_URL = "https://maps.googleapis.com/maps/api/place/";
+    private static final String GMAPS_KEY = System.getenv("GMAPS_API_KEY");
+    private static final WebClient webClient = WebClient.builder().build();
+
+    public MapsClient() {
+        // TODO probably put this somewhere else
+        if (GMAPS_KEY == null) {
+            throw new IllegalArgumentException("Please export environment variable GMAPS_KEY");
+        }
+    }
+
+
+    /**
+     * This function returns a list of the places near to the given coordinates and in a RADIUS defined on the class level.
+     * This will call the more generic endpoint of the GMaps API which is not adding some important info
+     * like opening hours, address, website etc.
+     * These can be added with the enriching function.
+     * Returns null if something goes wrong during the query.
+     */
+    public Optional<PlaceList> getNearbyPlaces(double coordX, double coordY) {
+        PlaceList places;
+        try {
+            places = webClient.get()
+                    .uri(GMAPS_BASE_URL + "nearbysearch/json?location="
+                            + coordX + "," + coordY
+                            + "&radius=" + RADIUS
+                            + "&key=" + GMAPS_KEY)
+                    .retrieve()
+                    .bodyToMono(PlaceList.class)
+                    .block();
```

Comment:
We shouldn't block in these calls. Just return a Mono instead of the Optional. You can add some error handling in the doOnError method.

Reply:
Completely agreed. Some tutorials were using `block` but I realised that blocking is basically going against the whole reactive philosophy.